goldenverba/components/generation/OpenAIGenerator.py

```
@@ -71,36 +103,111 @@ async def generate_stream(
+
+    async def _chat_completions_stream(self, headers, base_url, model, messages):
+        data = {"messages": messages, "model": model, "stream": True}
         async with httpx.AsyncClient() as client:
             async with client.stream(
                 "POST",
-                f"{openai_url}/chat/completions",
+                f"{base_url}/chat/completions",
                 json=data,
                 headers=headers,
                 timeout=None,
             ) as response:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)